pkg/repository/v1/trigger.go

```diff
@@ -153,27 +161,39 @@ func (r *TriggerRepositoryImpl) TriggerFromEvents(ctx context.Context, tenantId
 			}
 
 			additionalMetadata := triggerConverter.ToMetadata(opt.AdditionalMetadata)
+			externalId := uuid.NewString()
 
 			triggerOpts = append(triggerOpts, triggerTuple{
 				workflowVersionId:  sqlchelpers.UUIDToStr(workflow.WorkflowVersionId),
 				workflowId:         sqlchelpers.UUIDToStr(workflow.WorkflowId),
 				workflowName:       workflow.WorkflowName,
-				externalId:         uuid.NewString(),
+				externalId:         externalId,
 				input:              opt.Data,
 				additionalMetadata: additionalMetadata,
 			})
+
+			eventKeysTaskExternalIds = append(eventKeysTaskExternalIds, &EventDataForOlap{
+				EventKey:                workflow.EventKey,
+				TaskExternalId:          externalId,
+				EventAdditionalMetadata: additionalMetadata,
+				EventPayload:            opt.Data,
+			})
```

Comment:
wasn't sure about this but it seemed like a reasonable way to get the event data back to the message queue